Shopify/cli
# Propagate --verbose to oclif's logging

### WHY are these changes introduced?
The `DEBUG=*` environment variable has been quite handy in numerous time to debug issues coming from oclif. They were most of the times related to modules that failed to load. The problem with the variable is that it's not obvious that it exists. Moreover, there's already another flag that developers are already familiar with `--verbose` and serves a similar purpose.

### WHAT is this pull request doing?
I'm adjusting the function that triggers `oclif` to set the environment variable `DEBUG=*` when the `--verbose` flag is passed. Note that I had to make some changes in how the modules are loaded because static imports led to `debug`, the library that oclif uses for logging, loading and caching the value of the variable before we set it at runtime.

### How to test your changes?
Easy, run any Shopify command with `--verbose ` and you should see all the oclif logs.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
